### PR TITLE
Add force remove, rename Driver, update kontainer-engine

### PR DESCRIPTION
**Pull request by rmweir**
Problem: Removing unavailable cluster causes cluster to hang in removal state in UI.

Solution: Update to latest kontainer engine which logs errors caused by driver remove instead of returning it, then proceeds to remove cluster from persist store.

Issue: https://github.com/rancher/rancher/issues/18328

### CODE REVIEW

File: vendor/github.com/rancher/kontainer-engine/cluster/cluster.go [hunk 1]
```diff
@@ -305,7 +305,9 @@ func (c *Cluster) Remove(ctx context.Context) error {
 	}
 
 	if err := c.Driver.Remove(ctx, toInfo(c)); err != nil {
-		return err
+		// Persist store removal must take place despite error to prevent cluster from
+		// being stuck in remove state in UI
```
Comment (cjellick) [inline vendor/github.com/rancher/kontainer-engine/cluster/cluster.go]: Drop "in UI"
Add "TODO: We should add a "forceRemove" action to clusters and then revert this to returning an error, so that the user can see the problem and take appropriate action?

File: pkg/controllers/management/clusterprovisioner/provisioner.go [hunk 2]
```diff
@@ -37,7 +37,7 @@ type Provisioner struct {
 	ClusterController     v3.ClusterController
 	Clusters              v3.ClusterInterface
 	NodeLister            v3.NodeLister
-	Driver                service.EngineService
+	engineService         *service.EngineService
```
Comment (nathan-jenan-rancher) [inline pkg/controllers/management/clusterprovisioner/provisioner.go]: Why was this changed to a pointer?

**Comment by nathan-jenan-rancher** [comment]
LGTM

**Comment by cjellick** [comment]
LGTM

**Comment by Gregohr** [comment]
That fixed my problem. Thanks 👍💪
